xcode/X16/iOS/SettingsViewController.m

```diff
@@ -0,0 +1,77 @@
+//
+//  SettingsViewController.m
+//  CommanderX16
+//
+//  Created by Pearce, Matthew (Senior Developer) on 01/01/2020.
+//
+
+#import "SettingsViewController.h"
+#import "GameControlViewController.h"
+#include "SDL.h"
+#include "SDL_keyboard.h"
+
+@interface SettingsViewController ()
+
+-(IBAction)buttonPressed:(id)sender;
+-(IBAction)closeWindow:(id)sender;
+-(IBAction)toggleKeyboard:(id)sender;
+-(IBAction)showGamePad:(id)sender;
+
+@property (nonatomic, retain) GameControlViewController *game;
+@end
+
+@implementation SettingsViewController
+
+@synthesize game;
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    // Do any additional setup after loading the view from its nib.
+}
+
+/*
+#pragma mark - Navigation
+
+// In a storyboard-based application, you will often want to do a little preparation before navigation
+- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
+    // Get the new view controller using [segue destinationViewController].
+    // Pass the selected object to the new view controller.
+}
+*/
+
+-(IBAction)toggleKeyboard:(id)sender {
+
+    SDL_StartTextInput();
```

Comment:
You also do this in main(). Remove there? Remove here?

Reply:
You have to have a way of restarting input. If you completely dismiss the keyboard then SDL stops accepting input.